plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -165,6 +171,33 @@ static Settings getAvailabilityZoneNodeAttributes(Settings settings, String azMe
         return attrs.build();
     }
 
+    @SuppressForbidden(reason = "We call getInputStream in doPrivileged and provide SocketPermission")
+    private static Optional<String> getToken(String azMetadataTokenUrl) {
+        if (Strings.isNullOrEmpty(azMetadataTokenUrl)) {
+            return Optional.empty();
+        }
+        HttpURLConnection tokenUrlConnection;
+        try {
+            tokenUrlConnection = SocketAccess.doPrivilegedIOException(
```

Comment:
I guess we could include all the logic within this privileged block, that would simplify this code.

Reply:
That makes sense!